examples/set-maintenance

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+import os
+import re
+import logging
+from argparse import ArgumentParser
+
+from pubtools.pulplib import Client, Criteria, Matcher
+
+log = logging.getLogger("set-maintenance")
+
+
+def set_maintenance(client, regex):
+
+    report = client.get_maintenance_report().result()
+
+    crit = Criteria.with_field("id", Matcher.regex(regex))
+    # search repo
+    repos = self.search_repository(crit).result()
+
+    repo_ids = [repo.id for repo in repos.as_iter()]
+    report = report.add(repo_ids)
+    # get the repo ids and feed to report
+
+    tasks = client.set_maintainenance(report).result()
+
+    report = client.get_maintenance_report().result()
+    log.info("maintenance mode set: %s", report._json())
```

Comment:
Please avoid using the private API in the example.

Isn't the plain string representation giving you all the same data anyway, in arguably more readable form?